### PR TITLE
Move names into a public slice

### PR DESCRIPTION
**Pull request by virtualritz**
This can be used to contruct reverse lookups which I need to an app I am working on. Nothing else (API) was changed.

### CODE REVIEW

**Review by wezm — CHANGES_REQUESTED** [review]
Looks like there's some bits missing.

These files are missing:

- `aglfn_names.rs` (referenced in `lib.rs` and `Makefile`)
- `aglfn_names.awk` (referenced in `Makefile`)

I'm a little concerned by the diff on the `glyph_name` function:

`0x0020 => Some("space"), // SPACE` to `0x0020 => Some(GLYPH_NAME_PAIRS[0].1), // SPACE`. Previously this was generating a jump table now there is _potentially_ a bounds check on every lookup and the performance is important to us. It's possible that the compiler elides the bounds check but I'd like to verify that by looking at the assembly when you've added the missing files.

**Comment by virtualritz** [comment]
Heaps sorry. I was pulling an all nighter and forgot to add them.
As a bonus I also made `clippy` happy and moved `Cargo.toml` to Rust 2021 (also ran `cargo fix --edition` but that came out w. no changes).

**Comment by virtualritz** [comment]
> 0x0020 => Some("space"), // SPACE to 0x0020 => Some(GLYPH_NAME_PAIRS[0].1), // SPACE. Previously this was generating a jump table now there is potentially a bounds check on every lookup and the performance is important to us.

Fair enough. I'm curious about this too. A performance-safe solution would be to use:

```rust
0x0020 => Some(unsafe { GLYPH_NAME_PAIRS.get_unckecked(0).1 }),
```

That would mean the crate has `unsafe` code but it's provably fine since all tables and match blocks are autogenerated from the same file. Let me know if that is preferrable to you over spending time on `godbolt` and I add it ot the PR. :)

**Comment by wezm** [comment]
> That would mean the crate has `unsafe` code but it's provably fine since all tables and match blocks are autogenerated from the same file. Let me know if that is preferrable to you over spending time on `godbolt` and I add it ot the PR. :)

I'm not opposed to using `unsafe` since as you point out it would be auto-generated code known to only make in bounds accesses. Before doing that though I'd like to do some investigations to see what asm is actually being generated.

**Comment by wezm** [comment]
Well `cargo asm` reports identical assembly on `master` and this branch so doesn't look like there's a need to use `unsafe`.

File: Cargo.toml [hunk 1]
```diff
@@ -2,7 +2,7 @@
 name = "glyph-names"
 version = "0.1.0"
 authors = ["YesLogic Pty. Ltd. <info@yeslogic.com>"]
-edition = "2018"
+edition = "2021"
```
Comment (wezm) [inline Cargo.toml]: I would like this change and the `format!` shorthand to be reverted. It pushes up the minimum supported Rust version without providing any clear advantage to the codebase. There are some places where old versions of the compiler might still be in use (E.g. Debian stable) and we may as well remain compatible while there is little downside in the code.

**Review by wezm — CHANGES_REQUESTED** [review]

File: Cargo.toml (hunk 1 above)
Comment (virtualritz) [reply Cargo.toml]: Fair enough.

**Review by wezm — APPROVED** [review]
Looks good now, thanks 👍

**Pull request merged**